warehouse/query-core/src/main/java/datawave/query/predicate/ParentScanRangeProvider.java

```diff
@@ -0,0 +1,25 @@
+package datawave.query.predicate;
+
+import datawave.query.function.DocumentScanRangeProvider;
+import datawave.query.tld.TLD;
+import org.apache.accumulo.core.data.Key;
+import org.apache.hadoop.io.Text;
+
+/**
+ * For use with the {@link datawave.query.tables.ParentQueryLogic}
+ */
+public class ParentScanRangeProvider extends DocumentScanRangeProvider {
+    
+    /**
+     * Get the start key by parsing the root uid from the column family
+     * 
+     * @param k
+     *            an initial key
+     * @return the start key
+     */
+    @Override
+    public Key getStartKey(Key k) {
+        Text cf = new Text(TLD.parseRootPointerFromId(k.getColumnFamily().toString()));
```

Comment:
Wouldn't the parent start key be the document uid minus the last second, not the tld?

Reply:
I think this needs to change.

Reply:
Good catch -- there was also a bug in how the parent id was parsed. The ParentQueryLogic was actually a GrandParentQueryLogic